tasker/executors/sequential_executor.py

```
@@ -223,8 +225,23 @@ def execute_task(task, executor_instance):
                     executor_instance.log(f"Task {task_id}{loop_display}: Sleeping for {sleep_time} seconds")
-                    if not executor_instance.dry_run:
-                        time.sleep(sleep_time)
+                    if not executor_instance.dry_run and sleep_time > 0:
+                        # Use non-blocking sleep for consistency with parallel executor
+                        sleep_completion_event = threading.Event()
+
+                        def sleep_callback():
+                            executor_instance.log_debug(f"Task {task_id}{loop_display}: Sleep completed")
+                            sleep_completion_event.set()
+
+                        sleep_timer = sleep_async(
```

Ruff (0.13.2)

Local variable `sleep_timer` is assigned to but never used

Remove assignment to unused variable `sleep_timer`

(F841)